ipapython/install/core.py

```diff
@@ -189,6 +200,31 @@ def knob(type_=_missing, default=_missing, bases=_missing, sensitive=_missing,
     return util.InnerClassMeta('Knob', bases, class_dict)
 
 
+def knob(type, default=_missing, **kwargs):
+    return _knob(
+        type, default,
+        _order=next(_counter),
+        **kwargs
+    )
+
+
+def extend_knob(base, default=_missing, bases=_missing, group=_missing,
+                **kwargs):
```

Comment:
There shall be at least a short docstring for future humans who try to find out what happened here.